libs/openFrameworks/utils/ofFileUtils.cpp

```diff
@@ -1952,72 +1938,18 @@ std::string ofToDataPath(const of::filesystem::path & path, bool makeAbsolute){
 		}
 	}
 #endif
-
-	// this could be performed here, or wherever we might think we accidentally change the cwd, e.g. after file dialogs on windows
+	// FIXME: change to direct returns when return type is fs::path
+	of::filesystem::path outPath;
 	const auto & dataPath = dataPathRoot();
-	of::filesystem::path inputPath(path);
-	of::filesystem::path outputPath;
-
-	// if path is already absolute, just return it
-	if (inputPath.is_absolute()) {
-		try {
-			auto outpath = of::filesystem::canonical(inputPath).make_preferred();
-			if(of::filesystem::is_directory(outpath) && hasTrailingSlash){
-				return ofFilePath::addTrailingSlash(outpath);
-			}else{
-				return outpath.string();
-				// return outpath;
-			}
-		}
-		catch (...) {
-			return inputPath.string();
-			// return inputPath;
-		}
-	}
-
-	// here we check whether path already refers to the data folder by looking for common elements
-	// if the path begins with the full contents of dataPathRoot then the data path has already been added
-	// we compare inputPath.toString() rather that the input var path to ensure common formatting against dataPath.toString()
-	auto dirDataPath = dataPath;
-	// also, we strip the trailing slash from dataPath since `path` may be input as a file formatted path even if it is a folder (i.e. missing trailing slash)
-	dirDataPath = ofFilePath::addTrailingSlash(dataPath);
-
-	auto relativeDirDataPath = ofFilePath::makeRelative(of::filesystem::current_path(), dataPath);
-	relativeDirDataPath = ofFilePath::addTrailingSlash(relativeDirDataPath);
-
-	// FIXME: this can be simplified without using string conversion
-	// if (inputPath.string().find(dirDataPath.string()) != 0 && inputPath.string().find(relativeDirDataPath.string())!=0) {
-	if (inputPath.string().find(dirDataPath.string()) != 0 && inputPath.string().find(relativeDirDataPath)!=0) {
-		// inputPath doesn't contain data path already, so we build the output path as the inputPath relative to the dataPath
-		if(makeAbsolute){
-			outputPath = dirDataPath / inputPath;
-		}else{
-			outputPath = relativeDirDataPath / inputPath;
-		}
+	if (makeAbsolute) {
+		outPath = dataPath / path;
```

Comment:
Might be an edge case here if you pass in an absolute path and `makeAbsolute` is true. 
ie: path is `/Users/me/Desktop/image.jpg` if makeAbsolute is true and the path is already absolute then it would prepend dataPath ie: `/users/me/Downloads/OF/apps/myApp/bin/data/Users/Theo/Desktop/image.jpg`

does this edge case exist in the original code?  looks like it might?

Reply:
Great, I'll be investigating this better now